apps/nextjs/src/lib/github/client.ts

```
@@ -0,0 +1,211 @@
+
+  return {
+    accessToken: data.access_token,
+    username: userResponse.data.login,
+  };
+};
+
+export const listUserRepositories = async (accessToken: string) => {
+  const octokit = new Octokit({ auth: accessToken });
+  const response = await octokit.repos.listForAuthenticatedUser({
+    sort: "updated",
+    per_page: 100,
+  });
+
+  return response.data.map((repo) => ({
+    id: repo.id,
+    name: repo.name,
+    fullName: repo.full_name,
+    owner: repo.owner.login,
+    private: repo.private,
+  }));
+};
+
+export const checkRepositoryIssuesEnabled = async ({
+  accessToken,
+  owner,
+  repo,
+}: {
+  accessToken: string;
+  owner: string;
+  repo: string;
+}): Promise<boolean> => {
+  const octokit = new Octokit({ auth: accessToken });
+
+  try {
+    // Get repository details to check if issues are enabled
+    const { data } = await octokit.repos.get({
+      owner,
+      repo,
+    });
+
+    return data.has_issues === true;
+  } catch (error) {
+    console.error("Error checking repository issues status:", error);
+    throw new Error("Failed to check if issues are enabled for this repository");
+  }
+};
+
+export const listRepositoryIssues = async ({
+  accessToken,
+  owner,
+  repo,
+  state = "open",
+  per_page = 100,
+}: {
+  accessToken: string;
+  owner: string;
+  repo: string;
+  state?: "open" | "closed" | "all";
+  per_page?: number;
+}) => {
+  const octokit = new Octokit({ auth: accessToken });
+
+  try {
+    const { data } = await octokit.issues.listForRepo({
+      owner,
+      repo,
+      state,
+      per_page,
+      sort: "updated",
+      direction: "desc",
+    });
+
+    return data.map((issue) => ({
+      number: issue.number,
+      title: issue.title,
+      state: issue.state,
+      url: issue.html_url,
+      updatedAt: issue.updated_at,
+    }));
+  } catch (error) {
+    console.error("Error listing repository issues:", error);
+    throw new Error("Failed to list issues for this repository");
```

In general I think we can let things like this fail rather than having custom error handlers (if it's for the message, we should throw a tRPC error class in the procedure)

---

fix: [2e889c3](https://github.com/anti-work/helper/pull/47/commits/2e889c3c8298d7f5d0ec4951e578159ac0cfbab1)